information/dumps/objectconfigs.txt
('0','null','It is a null.'),('1','Crate','It is a Crate.'),('2','Cave Entrance','It is a Cave Entrance.'),('3','Door','It is a Door.'),('4','null','It is a null.'),('5','Broken multicannon','It is a Broken multicannon.'),('6','Dwarf multicannon','It is a Dwarf multicannon.'),('7','Cannon base','It is a Cannon base.'),('8','Cannon stand','It is a Cannon stand.'),('9','Cannon barrels','It is a Cannon barrels.'),('10','null','It is a null.'),('11','Ladder','It is a Ladder.'),('12','null','It is a null.'),('13','Mud pile','It is a Mud pile.'),('14','null','It is a null.'),('15','Railing','It is a Railing.'),('16','Railing','It is a Railing.'),('17','Railing','It is a Railing.'),('18','Railing','It is a Railing.'),('19','Railing','It is a Railing.'),('20','Railing','It is a Railing.'),('21','null','It is a null.'),('22','Door','It is a Door.'),('23','Sacks','It is a Sacks.'),('24','null','It is a null.'),('25','Clock spindle','It is a Clock spindle.'),('26','Clock spindle','It is a Clock spindle.'),('27','Clock spindle','It is a Clock spindle.'),('28','Clock spindle','It is a Clock spindle.'),('29','null','It is a null.'),('30','null','It is a null.'),('31','null','It is a null.'),('32','null','It is a null.'),('33','Lever','It is a Lever.'),('34','Lever','It is a Lever.'),('35','Lever','It is a Lever.'),('36','Lever','It is a Lever.'),('37','Gate','It is a Gate.'),('38','Gate','It is a Gate.'),('39','Gate','It is a Gate.'),('40','Food trough','It is a Food trough.'),('41','Wall Pipe','It is a Wall Pipe.'),('42','null','It is a null.'),('43','Water','It is a Water.'),('44','null','It is a null.'),('45','Staircase','It is a Staircase.'),('46','Staircase','It is a Staircase.'),('47','Gate','It is a Gate.'),('48','Gate','It is a Gate.'),('49','Gate','It is a Gate.'),('50','Gate','It is a Gate.'),('51','Loose Railing','It is a Loose Railing.'),('52','Gate','It is a Gate.'),('53','Gate','It is a Gate.'),('54','Stairs','It is a Stairs.'),('55','Stairs','It is a Stairs.'),('56','Stairs','It is a Stairs.'),('57','Stairs','It is a Stairs.'),('58','Vine','It is a Vine.'),('59','Door','It is a Door.'),('60','null','It is a null.'),('61','Chaos altar','It is a Chaos altar.'),('62','null','It is a null.'),('63','Crate','It is a Crate.'),('64','Crate','It is a Crate.'),('65','Crate wall','It is a Crate wall.'),('66','Crate wall','It is a Crate wall.'),('67','null','It is a null.'),('68','Beehive','It is a Beehive.'),('69','Gangplank','It is a Gangplank.'),('70','Gangplank','It is a Gangplank.'),('71','Large door','It is a Large door.'),('72','Large door','It is a Large door.'),('73','Large door','It is a Large door.'),('74','Large door','It is a Large door.'),('75','null','It is a null.'),('76','Parasol','It is a Parasol.'),('77','Door','It is a Door.'),('78','null','It is a null.'),('79','Prison door','It is a Prison door.'),('80','Prison door','It is a Prison door.'),('81','Door','It is a Door.'),('82','Door','It is a Door.'),('83','null','It is a null.'),('84','null','It is a null.'),('85','null','It is a null.'),('86','Lever bracket','It is a Lever bracket.'),('87','Lever','It is a Lever.'),('88','null','It is a null.'),('89','Gate','It is a Gate.'),('90','Gate','It is a Gate.'),('91','Lever','It is a Lever.'),('92','Door','It is a Door.'),('93','Door','It is a Door.'),('94','Gate','It is a Gate.'),('95','Gate','It is a Gate.'),('96','Stairs','It is a Stairs.'),('97','null','It is a null.'),('98','null','It is a null.'),('99','Door','It is a Door.'),('100','Trapdoor','It is a Trapdoor.'),('101','Ladder','It is a Ladder.'),('102','Door','It is a Door.'),('103','null','It is a null.'),('104','null','It is a null.'),('105','Trapdoor','It is a Trapdoor.'),('106','Trapdoor','It is a Trapdoor.'),('107','null','It is a null.'),('108','null','It is a null.'),('109','null','It is a null.'),('110','null','It is a null.'),('111','null','It is a null.'),('112','Lava engine','It is a Lava engine.'),('113','null','It is a null.'),('114','Cooking range','It is a Cooking range.'),('115','Party Balloon','It is a Party Balloon.'),('116','Party Balloon','It is a Party Balloon.'),('117','Party Balloon','It is a Party Balloon.'),('118','Party Balloon','It is a Party Balloon.'),('119','Party Balloon','It is a Party Balloon.'),('120','Party Balloon','It is a Party Balloon.'),('121','Party Balloon','It is a Party Balloon.'),('122','Party Balloon','It is a Party Balloon.'),('123','Party Balloon','It is a Party Balloon.'),('124','Party Balloon','It is a Party Balloon.'),('125','Party Balloon','It is a Party Balloon.'),('126','Party Balloon','It is a Party Balloon.'),('127','Party Balloon','It is a Party Balloon.'),('128','Party Balloon','It is a Party Balloon.'),('129','Party Balloon','It is a Party Balloon.'),('130','Party Balloon','It is a Party Balloon.'),('131','Door','It is a Door.'),('132','Bar','It is a Bar.'),('133','Ladder','It is a Ladder.'),('134','null','It is a null.'),('135','null','It is a null.'),('136','Bar','It is a Bar.'),('137','null','It is a null.'),('138','null','It is a null.'),('139','Willow','It is a Willow.'),('140','null','It is a null.'),('141','null','It is a null.'),('142','Willow','It is a Willow.'),('143','null','It is a null.'),('144','null','It is a null.'),('145','null','It is a null.'),('146','null','It is a null.'),('147','null','It is a null.'),('148','null','It is a null.'),('149','null','It is a null.'),('150','null','It is a null.'),('151','null','It is a null.'),('152','Compost heap','It is a Compost heap.'),('153','Fountain','It is a Fountain.'),('154','null','It is a null.'),('155','null','It is a null.'),('156','null','It is a null.'),('157','Bar','It is a Bar.'),('158','Coffin','It is a Coffin.'),('159','Coffin','It is a Coffin.'),('160','Lever','It is a Lever.'),('161','null','It is a null.'),('162','Coffin','It is a Coffin.'),('163','Coffin','It is a Coffin.'),('164','Stairs','It is a Stairs.'),('165','Incinerator','It is a Incinerator.'),('166','Gate','It is a Gate.'),('167','Gate','It is a Gate.'),('168','null','It is a null.'),('169','null','It is a null.'),('170','Closed chest','It is a Closed chest.'),('171','Open chest','It is a Open chest.'),('172','Crystal chest','It is a Crystal chest.'),('173','Crystal chest','It is a Crystal chest.'),('174','null','It is a null.'),('175','null','It is a null.'),('176','null','It is a null.'),('177','null','It is a null.'),('178','null','It is a null.'),('179','null','It is a null.'),('180','null','It is a null.'),('181','null','It is a null.'),('182','null','It is a null.'),('183','Mine entrance','It is a Mine entrance.'),('184','null','It is a null.'),('185','null','It is a null.'),('186','null','It is a null.'),('187','Gnome glider','It is a Gnome glider.'),('188','Crashed glider','It is a Crashed glider.'),('189','Blurberry bar','It is a Blurberry bar.'),('190','Gate','It is a Gate.'),('191','null','It is a null.'),('192','null','It is a null.'),('193','null','It is a null.'),('194','Hollowed rock','It is a Hollowed rock.'),('195','Ladder','It is a Ladder.'),('196','null','It is a null.'),('197','null','It is a null.'),('198','null','It is a null.'),('199','null','It is a null.'),('200','Lamp','It is a Lamp.'),('201','Lamp','It is a Lamp.'),('202','Candles','It is a Candles.'),('203','Candles','It is a Candles.'),('204','Candles','It is a Candles.'),('205','null','It is a null.'),('206','null','It is a null.'),('207','null','It is a null.'),('208','Candles','It is a Candles.'),('209','null','It is a null.'),('210','null','It is a null.'),('211','null','It is a null.'),('212','Sail','It is a Sail.'),('213','Wall','It is a Wall.'),('214','Wall','It is a Wall.'),('215','Wall','It is a Wall.'),('216','Wall','It is a Wall.'),('217','Ruined rope rack','It is a Ruined rope rack.'),('218','Sail','It is a Sail.'),('219','Rope rack','It is a Rope rack.'),('220','null','It is a null.'),('221','null','It is a null.'),('222','null','It is a null.'),('223','null','It is a null.'),('224','Rock','It is a Rock.'),('225','null','It is a null.'),('226','null','It is a null.'),('227','Pile of rope','It is a Pile of rope.'),('228','null','It is a null.'),('229','null','It is a null.'),('230','null','It is a null.'),('231','null','It is a null.'),('232','null','It is a null.'),('233','null','It is a null.'),('234','null','It is a null.'),('235','null','It is a null.'),('236','null','It is a null.'),('237','null','It is a null.'),('238','null','It is a null.'),('239','null','It is a null.'),('240','null','It is a null.'),('241','Strong tree','It is a Strong tree.'),('242','null','It is a null.'),('243','null','It is a null.'),('244','Strong tree','It is a Strong tree.'),('247','null','It is a null.'),('248','null','It is a null.'),('249','null','It is a null.'),('250','Strong tree','It is a Strong tree.'),('251','Grapple column','It is a Grapple column.'),('253','null','It is a null.'),('254','Figurehead','It is a Figurehead.'),('255','Winch','It is a Winch.'),('256','null','It is a null.'),('257','null','It is a null.'),('258','null','It is a null.'),('259','null','It is a null.'),('260','Figurehead','It is a Figurehead.'),('261','null','It is a null.'),('262','null','It is a null.'),('263','null','It is a null.'),('264','null','It is a null.'),('265','Sail','It is a Sail.'),('266','Sail','It is a Sail.'),('267','Sail','It is a Sail.'),('268','Sail','It is a Sail.'),('269','null','It is a null.'),('270','null','It is a null.'),('271','Anchor','It is a Anchor.'),('274','Hoisted sail','It is a Hoisted sail.'),('275','Hoisted sail','It is a Hoisted sail.'),('276','Hoisted sail','It is a Hoisted sail.'),('277','Hoisted sail','It is a Hoisted sail.'),('278','Hoisted sail','It is a Hoisted sail.'),('279','Hoisted sail','It is a Hoisted sail.'),('280','null','It is a null.'),('281','null','It is a null.'),('282','Convenient rock','It is a Convenient rock.'),('283','null','It is a null.'),('284','null','It is a null.'),('285','null','It is a null.'),('286','null','It is a null.'),('287','Ladder','It is a Ladder.'),('288','Convenient rock','It is a Convenient rock.'),('289','null','It is a null.'),('290','null','It is a null.'),('291','null','It is a null.'),('292','null','It is a null.'),('293','Convenient rock','It is a Convenient rock.'),('294','null','It is a .'),('295','null','It is a null.'),('296','Plough','It is a Plough.'),('297','Scarecrow','It is a Scarecrow.'),('298','Hay bales','It is a Hay bales.'),('299','Hay bales','It is a Hay bales.'),('300','Haystack','It is a Haystack.'),('301','Food trough','It is a Food trough.'),('302','null','It is a null.'),('303','Compost','It is a Compost.'),('304','Hay bale','It is a Hay bale.'),('305','null','It is a null.'),('306','Broken cart','It is a Broken cart.'),('307','Cart','It is a Cart.'),('308','Cart','It is a Cart.'),('309','Log pile','It is a Log pile.'),('310','null','It is a null.'),('311','null','It is a null.'),('312','Potato','It is a Potato.'),('313','Wheat','It is a Wheat.'),('314','null','It is a null.'),('315','null','It is a null.'),('316','null','It is a null.'),('317','null','It is a null.'),('318','null','It is a null.'),('319','null','It is a null.'),('320','null','It is a null.'),('321','null','It is a null.'),('322','null','It is a null.'),('323','null','It is a null.'),('324','null','It is a null.'),('325','null','It is a null.'),('326','null','It is a null.'),('327','Broken cart wheel','It is a Broken cart wheel.'),('328','null','It is a null.'),('329','null','It is a null.'),('330','null','It is a null.'),('331','null','It is a null.'),('332','null','It is a null.'),('333','null','It is a null.'),('334','null','It is a null.'),('335','null','It is a null.'),('336','null','It is a null.'),('337','null','It is a null.'),('338','null','It is a null.'),('339','null','It is a null.'),('340','null','It is a null.'),('341','null','It is a null.'),('342','null','It is a null.'),('343','null','It is a null.'),('344','null','It is a null.'),('345','null','It is a null.'),('346','Cabinet','It is a Cabinet.'),('347','Cabinet','It is a Cabinet.'),('348','Drawers','It is a Drawers.'),('349','Drawers','It is a Drawers.'),('350','Drawers','It is a Drawers.'),('351','Drawers','It is a Drawers.'),('352','Drawers','It is a Drawers.'),('353','Drawers','It is a Drawers.'),('354','Crate','It is a Crate.'),('355','Crate','It is a Crate.'),('356','Crate','It is a Crate.'),('357','Crate','It is a Crate.'),('358','Crate','It is a Crate.'),('359','Boxes','It is a Boxes.'),('360','Boxes','It is a Boxes.'),('361','Boxes','It is a Boxes.'),('362','Barrel','It is a Barrel.'),('363','Barrel','It is a Barrel.'),('364','Barrel','It is a Barrel.'),('365','Sacks','It is a Sacks.'),('366','Crate','It is a Crate.'),('367','Cupboard','It is a Cupboard.'),('368','Cupboard','It is a Cupboard.'),('369','Large cupboard','It is a Large cupboard.'),('370','Open Cupboard','It is a Open Cupboard.'),('371','Cupboard','It is a Cupboard.'),('372','Large open cupboard','It is a Large open cupboard.'),('373','Dresser','It is a Dresser.'),('374','Hat stand','It is a Hat stand.'),('375','Closed chest','It is a Closed chest.'),('376','Closed chest','It is a Closed chest.'),('377','Closed chest','It is a Closed chest.'),('378','Open chest','It is a Open chest.'),('379','Open chest','It is a Open chest.'),('380','Bookcase','It is a Bookcase.'),('381','Bookcase','It is a Bookcase.'),('382','Basket','It is a Basket.'),('383','Basket','It is a Basket.'),('384','null','It is a null.'),('385','Basket','It is a Basket.'),('386','Large urn','It is a Large urn.'),('387','Bookshelves','It is a Bookshelves.'),('388','Wardrobe','It is a Wardrobe.'),('389','Wardrobe','It is a Wardrobe.'),('390','Wardrobe','It is a Wardrobe.'),('391','null','It is a null.'),('392','Full fishing net','It is a Full fishing net.'),('393','Bookcase','It is a Bookcase.'),('394','Bookcase','It is a Bookcase.'),('395','Pot','It is a Pot.'),('396','Pots','It is a Pots.'),('397','null','It is a null.'),('398','Coffin','It is a Coffin.'),('399','Coffin','It is a Coffin.'),('400','Small gravestone','It is a Small gravestone.'),('401','Grave marker','It is a Grave marker.'),('402','Tombstone','It is a Tombstone.'),('403','null','It is a null.'),('404','Gravestone','It is a Gravestone.'),('405','Gravestone','It is a Gravestone.'),('406','Large grave','It is a Large grave.'),('407','null','It is a null.'),('408','null','It is a null.'),('409','Altar','It is a Altar.'),('410','null','It is a null.'),('411','Chaos altar','It is a Chaos altar.'),('412','Chaos altar','It is a Chaos altar.'),('413','Dolmen','It is a Dolmen.'),('414','Coffin','It is a Coffin.'),('415','Coffin','It is a Coffin.'),('416','Church organ','It is a Church organ.'),('417','Bed','It is a Bed.'),('418','Bed','It is a Bed.'),('419','Bed','It is a Bed.'),('420','null','It is a null.'),('421','Bed','It is a Bed.'),('422','Bed','It is a Bed.'),('423','Bed','It is a Bed.'),('424','Bed','It is a Bed.'),('425','Bed','It is a Bed.'),('426','Bed','It is a Bed.'),('427','Bed','It is a Bed.'),('428','null','It is a null.'),('429','Bed','It is a Bed.'),('430','Hammock','It is a Hammock.'),('431','Chaise lounge','It is a Chaise lounge.'),('432','Bed','It is a Bed.'),('433','Bed','It is a Bed.'),('434','Bed','It is a Bed.'),('435','Bed','It is a Bed.'),('436','Rock','It is a Rock.'),('437','null','It is a null.'),('438','Rock','It is a Rock.'),('439','Rock','It is a Rock.'),('440','Rock','It is a Rock.'),('441','Rock','It is a Rock.'),('442','Rock','It is a Rock.'),('443','null','It is a null.'),('444','Boulder','It is a Boulder.'),('445','Boulder','It is a Boulder.'),('446','Stones','It is a Stones.'),('447','Stones','It is a Stones.'),('448','null','It is a null.'),('449','Overhanging rock','It is a Overhanging rock.'),('450','Rocks','It is a Rocks.'),('451','Rocks','It is a Rocks.'),('452','Rocks','It is a Rocks.'),('453','Rocks','It is a Rocks.'),('454','null','It is a null.'),('455','null','It is a null.'),('456','null','It is a null.'),('457','null','It is a null.'),('458','null','It is a null.'),('459','null','It is a null.'),('460','null','It is a null.'),('461','null','It is a null.'),('462','Mossy rock','It is a Mossy rock.'),('463','Mossy rock','It is a Mossy rock.'),('464','Column','It is a Column.'),('465','Column','It is a Column.'),('466','Column','It is a Column.'),('467','null','It is a null.'),('468','null','It is a null.'),('469','null','It is a null.'),('470','Ivy','It is a Ivy.'),('471','Rockslide','It is a Rockslide.'),('472','Rockslide','It is a Rockslide.'),('473','Rockslide','It is a Rockslide.'),('474','Rockslide','It is a Rockslide.'),('475','null','It is a null.'),('476','null','It is a null.'),('477','Mossy rock','It is a Mossy rock.'),('478','Mossy rock','It is a Mossy rock.'),('479','null','It is a null.'),('480','null','It is a null.'),('481','Rockslide','It is a Rockslide.'),('482','null','It is a null.'),('483','Rockslide','It is a Rockslide.'),('484','Rockslide','It is a Rockslide.'),('485','Rockslide','It is a Rockslide.'),('486','Rockslide','It is a Rockslide.'),('487','null','It is a null.'),('488','null','It is a null.'),('489','null','It is a null.'),('490','null','It is a null.'),('491','null','It is a null.'),('492','Rocks','It is a Rocks.'),('493','null','It is a null.'),('494','null','It is a null.'),('495','null','It is a null.'),('496','null','It is a null.'),('497','null','It is a null.'),('498','null','It is a null.'),('499','null','It is a null.'),('500','null','It is a null.'),('501','null','It is a null.'),('502','null','It is a null.'),('503','null','It is a null.'),('504','Hollow log','It is a Hollow log.'),('505','Ladder','It is a Ladder.'),('506','Scaffold','It is a Scaffold.'),('507','Rockslide','It is a Rockslide.'),('508','Rockslide','It is a Rockslide.'),('509','Rockslide','It is a Rockslide.'),('510','Rockslide','It is a Rockslide.'),('511','null','It is a null.'),('512','null','It is a null.'),('513','null','It is a null.'),('514','null','It is a null.'),('515','null','It is a null.'),('516','null','It is a null.'),('517','Rockslide','It is a Rockslide.'),('518','Rockslide','It is a Rockslide.'),('519','Rockslide','It is a Rockslide.'),('520','Rockslide','It is a Rockslide.'),('521','null','It is a null.'),('522','null','It is a null.'),('523','Rocks','It is a Rocks.'),('524','null','It is a null.'),('525','null','It is a null.'),('526','null','It is a null.'),('527','Trapdoor','It is a Trapdoor.'),('528','Stairs','It is a Stairs.'),('529','Bookcase','It is a Bookcase.'),('530','Bookcase','It is a Bookcase.'),('531','Bookcase','It is a Bookcase.'),('532','Stalagmites','It is a Stalagmites.'),('533','Bookcase','It is a Bookcase.'),('534','Stalagmite','It is a Stalagmite.'),('535','null','It is a null.'),('536','null','It is a null.'),('537','null','It is a null.'),('538','null','It is a null.'),('539','Bookcase','It is a Bookcase.'),('540','Workbench','It is a Workbench.'),('541','Stalagmite','It is a Stalagmite.'),('542','Stalagmite','It is a Stalagmite.'),('543','Stalactite','It is a Stalactite.'),('544','null','It is a null.'),('545','Animation table','It is a Animation table.'),('546','Animation table','It is a Animation table.'),('547','Animation table','It is a Animation table.'),('548','Rockslide','It is a Rockslide.'),('549','Rockslide','It is a Rockslide.'),('550','Rockslide','It is a Rockslide.'),('551','Rockslide','It is a Rockslide.'),('552','null','It is a null.'),('553','null','It is a null.'),('554','null','It is a null.'),('555','Icicle','It is a Icicle.'),('556','null','It is a null.'),('557','null','It is a null.'),('558','null','It is a null.'),('559','null','It is a null.'),('560','null','It is a null.'),('561','null','It is a null.'),('562','Statue of a warrior','It is a Statue of a warrior.'),('563','Statue of a king','It is a Statue of a king.'),('564','null','It is a null.'),('565','null','It is a null.'),('566','Statue of a warrior','It is a Statue of a warrior.'),('567','null','It is a null.'),('568','Dwarf statue','It is a Dwarf statue.'),('569','Gnome statue','It is a Gnome statue.'),('570','Gnome statue','It is a Gnome statue.'),('571','Gnome statue','It is a Gnome statue.'),('572','Gnome statue','It is a Gnome statue.'),('573','Gnome statue','It is a Gnome statue.'),('574','Statue','It is a Statue.'),('575','Statue','It is a Statue.'),('576','Statue','It is a Statue.'),('577','null','It is a null.'),('578','null','It is a null.'),('579','Statue','It is a Statue.'),('580','Statue','It is a Statue.'),('581','null','It is a null.'),('582','Statue','It is a Statue.'),('583','Vase','It is a Vase.'),('584','Statue','It is a Statue.'),('585','Statue of Saradomin','It is a Statue of Saradomin.'),('586','null','It is a null.'),('587','Statue of Zamorak','It is a Statue of Zamorak.'),('588','Plinth','It is a Plinth.'),('589','Crystal ball','It is a Crystal ball.'),('590','Bank table','It is a Bank table.'),('591','Bank table','It is a Bank table.'),('592','Bathroom table','It is a Bathroom table.'),('593','Table','It is a Table.'),('594','Table','It is a Table.'),('595','Table','It is a Table.'),('596','Table','It is a Table.'),('597','Table','It is a Table.'),('598','Table','It is a Table.'),('599','Table','It is a Table.'),('600','Table','It is a Table.'),('601','Table','It is a Table.'),('602','Table','It is a Table.'),('603','Table','It is a Table.'),('604','Table','It is a Table.'),('605','Table','It is a Table.'),('606','Table','It is a Table.'),('607','Table','It is a Table.'),('608','Study desk','It is a Study desk.'),('609','Workbench','It is a Workbench.'),('610','Workbench','It is a Workbench.'),('611','Picnic bench','It is a Picnic bench.'),('612','Counter','It is a Counter.'),('613','null','It is a null.'),('614','Table','It is a Table.'),('615','null','It is a null.'),('616','Table','It is a Table.'),('617','Plant','It is a Plant.'),('618','Chemistry Table','It is a Chemistry Table.'),('619','Chemistry Table','It is a Chemistry Table.'),('620','Table','It is a Table.'),('621','Table','It is a Table.'),('622','null','It is a null.'),('623','Frame','It is a Frame.'),('624','Table','It is a Table.'),('625','null','It is a null.'),('626','Frame','It is a Frame.'),('627','null','It is a null.'),('628','null','It is a null.'),('629','Silk stall','It is a Silk stall.'),('630','null','It is a null.'),('631','null','It is a null.'),('632','Fur stall','It is a Fur stall.'),('633','null','It is a null.'),('634','Market stall','It is a Market stall.'),('635','Tea stall','It is a Tea stall.'),('636','null','It is a null.'),('637','Mine cart','It is a Mine cart.'),('638','null','It is a null.'),('639','null','It is a null.'),('640','null','It is a null.'),('641','null','It is a null.'),('642','null','It 

src/net/nocturne/Settings.java
package net.nocturne;

import java.io.File;
import java.io.IOException;
import java.math.BigInteger;
import java.net.InetAddress;
import java.net.InetSocketAddress;
import java.net.MalformedURLException;
import java.net.UnknownHostException;
import java.util.ArrayList;

import net.nocturne.game.WorldTile;
import net.nocturne.game.player.Player;
import net.nocturne.login.WorldInformation;

public final class Settings {

	private static boolean hostName() {
		try {
			switch (InetAddress.getLocalHost().getHostName()) {
			case "nocturne":
				return true;
			}
		} catch (UnknownHostException e) {
			e.printStackTrace();
		}
		return false;
	}

	public static int WORLD_ID = 1;
	public static boolean DEBUG = hostName() ? false : true;
	public static boolean HOSTED = hostName() ? true : false;

	public static final String VPS1_IP = HOSTED ? "104.155.26.178"
			: "127.0.0.1";

	public static final String UPDATE_TOPIC_TITLE = "";
	public static final String UPDATE_TOPIC_ID = "";
	public static final String SERVER_NAME = "Nocturne";
	public static final String MAIN_FC = "danny";
	public static final String CACHE_PATH = "./data/cache/";
	public static final String LOGIN_DATA_PATH = "data/accounts_data";
	@SuppressWarnings("serial")
	public static final ArrayList<String> SERVER_OWNERS = new ArrayList<String>() {
		{
			add("danny");
		}
	};
	@SuppressWarnings("serial")
	public static final ArrayList<String> SERVER_DEVELOPERS = new ArrayList<String>() {
		{
			add("phil");
			add("tom");
		}
	};
	@SuppressWarnings("serial")
	public static final ArrayList<String> SERVER_ADMINISTRATORS = new ArrayList<String>() {
		{
			add("liam");
			add("klein_curse");
			add("spoyder");
		}
	};
	@SuppressWarnings("serial")
	public static final ArrayList<String> SERVER_MODERATORS = new ArrayList<String>() {
		{
			add("abigail");
			add("nathaniwan");
			add("kurain");
			add("nexus");
			add("toxic");
		}
	};

	public static final int XP_RATE = 1;
	public static boolean DOUBLE_XP = false;
	public static boolean DOUBLE_VOTES = false;
	private static final String MASTER_IP = "127.0.0.1";
	public static final WorldTile HOME_LOCATION = new WorldTile(1822, 11615, 0);
	public static final WorldTile STARTER_LOCATION = new WorldTile(1854, 11637,
			0);
	public final static int GAME_UPDATES = 1, WEBSITE = 2, SUPPORT = 3,
			TECHNICAL = 4, COMMUNITY = 5, BEHIND_THE_SCENES = 6, SHOP = 9,
			FUTURE_UPDATES = 12, SOLOMANS_STORE = 13, TREASURE_HUNTER = 14,
			YOUR_FEEDBACK = 15, EVENTS = 16;

	public static final boolean SQL = false;
	public static final boolean HTTP = false;
	public static final boolean DAILY_BACKUP = false;

	public static final int STARTER_GUIDE = 0;
	public static final String WEBSITE_LINK = "http://www.nocturne3.org";
	public static final String FORUMS_LINK = "http://www.nocturne3.org/community";
	public static final String BUG_LINK = "https://goo.gl/forms/jC0qSRam5XkEfr0l1";
	public static final String DISCORD_LINK = "https://discord.gg/GqCXPUY";
	public static final String HIGHSCORES_LINK = "http://www.nocturne3.org/highscores.php";
	public static final String VOTE_LINK = "http://www.nocturne3.org/vote";
	public static final String STORE_LINK = "http://rsps-pay.com/store.php?id=254";
	public static final String RULES_LINK = "http://nocturne3.org/community/index.php?threads/updated-official-server-rules.1316";
	public static final String REGISTER_LINK = "http://nocturne3.org/community/index.php?register";
	public static final String VOTE_HELP = "https://nocturne3.org/community/index.php?/topic/173-officialvoting-you/";
	public static final String[] RARE_DROPS = { "pernix", "torva", "virtus",
			"abyssal", "dark bow", "bandos", "steadfast", "glaiven",
			"ragefire", "spirit shield", "dragon claw", "berserker ring",
			"warrior ring", "archers' ring", "seers' ring", "hilt",
			"saradomin", "armadyl", "subjugation", "drygore",
			"draconic visage", "ascension", "tetsu", "death lotus",
			"seasinger's", "spider leg", "araxxi", "araxyte egg",
			"dragon pickaxe", "partyhat", "party hat", "cracker",
			"dragon hatchet", "sigil", "wand of treachery", "abyssal whip",
			"whip vine", "spirit shield", "seismic", "crest", "anima",
			"dragon rider", "shadow glaive", "cywir elders", "nymora",
			"avaryss", "zamorakian spear", "zaryte", "celestial handwrap",
			"razorback gauntlet", "blood necklace", "tetsu", "singer",
			"death lotus" };

	public static final String[] ANNOUNCEMENT_TEXTS = new String[] {

			"We are currently Live! Thank you for the support the past few months!",
			"If you have any suggestions please let us know on discord via our suggestions & polls channels.",
			"Nocturne RS3 is now live! please report any bugs by doing ;;report.",
			"The donation system is now live, ;;donate for more info!",
			"Help us fix bugs by reporting them with ::report.",
			"You can now teleport to skilling locations by clicking on a skill in the skills tab!",
			"Combine your strange rocks to access the endgame boss: Lucien.",
			"Want to keep updated? ::discord to stay up to date with the latest!",
			"Use game point rewards to improve efficiency, use ;;gpr for more info.",
			"See our FAQ on our Discord before asking any questions, join using ;;discord.",
			"Need help? Join 'danny' FC for community assistance!",
			"We have a wiki page for guides and info! Access it using ::wikia.",
			"Trouble voting? type ;;votehelp for a tutorial on how to vote correctly!" };

	public static final int MAJOR_VERSION = 876;
	public static final int MINOR_VERSION = 1;
	public static final int PACKET_SIZE_LIMIT = 15000;
	public static final int READ_BUFFER_SIZE = 200 * 1024; // 200kb
	public static final int WRITE_BUFFER_SIZE = 200 * 1024; // 200kb
	public static final int WORLD_CYCLE_TIME = 600; // the speed of world in ms
	public static final int[] MAP_SIZES = { 104, 120, 136, 168, 72 };
	public static final int PLAYERS_LIMIT = 2000;
	public static final int NPCS_LIMIT = Short.MAX_VALUE;
	public static final int LOCAL_NPCS_LIMIT = 400;
	static final int MIN_FREE_MEM_ALLOWED = 30000000; // 30mb
	public static final long LOGIN_SERVER_RETRY_DELAY = 1000;
	public static final long LOGIN_SERVER_FILE_TIMEOUT = 2000;
	public static final long LOGIN_SERVER_REQUEST_TIMEOUT = 3000;
	public static final long LOGIN_AUTOSAVE_INTERVAL = 500 * 60; // Default 1000
																	// * 60;
	public static final long LOGIN_BLOCKER_RESET_TIME = 1000 * 60 * 5;
	public static final int LOGIN_BLOCKER_MINIMUM_COUNT = 5;
	public static final long LOGIN_OFFENCES_CHECK_INTERVAL = 1000 * 60 * 30;
	public static final long LOGIN_FRIEND_CHATS_CHECK_INTERVAL = 1000 * 60;
	public static final String GRAB_SERVER_TOKEN = "fngDR7vSeguoCXAGiQM*DFVGv6SrfUbb";

	public static final String WORLD_SERVER_TOKEN = "3F110CFC079B70003DDFA581F69AF06E8D34A1418C593CC61BBE7C986041C46BC75A1074E9BCBCC0C15E5034192981D2";

	public static final String CLIENT_SETTINGS = "Ymi9Liy-EKAQPDzvIFDOawIZao205SgIjSoemJvZ5wo";

	public static final int CLIENT_LOGIN_ID = 1241012674;

	public static final int[] GRAB_SERVER_KEYS = { 3269, 69795, 41651, 35866,
			358716, 44375, 18239, 22668, 209410, 1155815, 463294, 597315,
			932750, 1247062, 37288, 920969, 1856, 21550, 1244, 76028, 2242,
			119, 1435171, 4510117, 9084, 23487 };

	public static final BigInteger GRAB_SERVER_PRIVATE_EXPONENT = new BigInteger(
			"37952323194930410014007478577313595429535985925786579609940606712850553335094656773153776498007258071864762773257091910728408524263381649029478057988929");
	public static final BigInteger GRAB_SERVER_MODULUS = new BigInteger(
			"7871143687424538864202557352915826277422468068412262873090751715633185803551126047872098304035825036280420995543699421150200902637147867439970995255532811");

	public static final BigInteger MODULUS = GRAB_SERVER_MODULUS;
	public static final BigInteger PRIVATE_EXPONENT = GRAB_SERVER_PRIVATE_EXPONENT;

	public static InetSocketAddress LOGIN_SERVER_ADDRESS_BASE;
	public static InetSocketAddress LOGIN_CLIENT_ADDRESS_BASE;
	public static WorldInformation[] WORLDS_INFORMATION;
	public static final InetSocketAddress GAME_ADDRESS_BASE = new InetSocketAddress(
			Settings.HOSTED ? "10.132.0.2" : "127.0.0.1", 43593);

	public static boolean underDevelopment(Player player) {
		if (!Settings.DEBUG) {
			player.getDialogueManager().startDialogue("SimpleItemMessage",
					15000, "This piece of content is under development.");
			return true;
		}
		return false;
	}

	static void init() {
		LOGIN_SERVER_ADDRESS_BASE = new InetSocketAddress(
				HOSTED ? "10.132.0.2" : "127.0.0.1", 7777);
		LOGIN_CLIENT_ADDRESS_BASE = new InetSocketAddress(
				HOSTED ? "10.132.0.2" : "127.0.0.1", 7778);
		
			WORLDS_INFORMATION = new WorldInformation[] {
					new WorldInformation(1, 0, "World1", 0, 0x2 | 0x8, "Nocturne :D", "10.132.0.2", 100),
					new WorldInformation(2, 0, "World2", 0, 0x2 | 0x8, "Nocturne :P", VPS1_IP, 100),};
	}

	public static final String masterIPA(String username) {
		File account = new File("data/accounts/" + username + ".acc");
		if (!account.exists())
			return ".";
		return MASTER_IP;
	}

	public static double getLampXpRate() {
		return 10;
	}

	public static String getDropboxLocation() {
		String host = "null";
		try {
			host = InetAddress.getLocalHost().getHostName();
		} catch (UnknownHostException e) {
			e.printStackTrace();
		}
		return host.equalsIgnoreCase("Miles") ? "D:/Dropbox/" : System
				.getProperty("user.home") + "/RS3Dropbox/";
	}

}